File: Quick Sort/quicksort.py
def quick_sort(arr):
    if len(arr) <= 1:
        return arr

    pivot = arr[-1]
    j = 0
    i = j - 1

    while j < len(arr) - 1:
        if arr[j] < pivot:
            i += 1
            temp = arr[i]
            arr[i] = arr[j]
            arr[j] = temp
        j += 1
    
    i += 1
    temp = arr[i]
    arr[i] = pivot
    arr[-1] = temp

    left_sub_array = quick_sort(arr[0:i])
    rigth_sub_array = quick_sort(arr[i + 1:])

    return left_sub_array + [pivot] + rigth_sub_array

File: Quick Sort/quicksort.js
function quickSort(arr) {
    if (arr.length <= 1){
        return arr
    }
    
    var pivot = arr[arr.length - 1];
    var j = 0
    var i = j - 1
    while (j < arr.length - 1) {
        if (arr[j] < pivot) {
            i += 1
            var temp = arr[i]
            arr[i] = arr[j]
            arr[j] = temp
        }
        j += 1
    }
    i += 1
    var temp = arr[i]
    arr[i] = pivot
    arr[arr.length - 1] = temp

    var left_sub_array = quickSort(arr.slice(0, i))
    var right_sub_array = quickSort(arr.slice(i + 1, arr.length))

    return left_sub_array.concat([pivot].concat(right_sub_array))
}
